msteams-platform/messaging-extensions/build-api-based-message-extension.md

```diff
@@ -73,14 +111,150 @@ If the root object of the OpenAPI schema contains well-known array property name
+
+# [None](#tab/none)
+
+
+
+# [API service auth](#tab/api-service-auth)
+
+API secret service authentication is a secure method for your app to authenticate with API. You can authorize incoming requests to your service by configuring a static API key. The API key is stored securely and added to the API call by the bot. Add an `apiSecretServiceAuthConfiguration` object with an `apiSecretRegistrationId` property, which contains the reference ID returned when you submitted the API key through the Developer portal for Teams.
+
+```json
+"composeExtensions": [
+    {
+      "composeExtensionType": "apiBased",
+      "authorization": {
+        "authType": "apiSecretServiceAuth ",
+        "apiSecretServiceAuthConfiguration": {
+            "apiSecretRegistrationId": "96270b0f-7298-40cc-b333-152f84321813"
+        }
+      },
+```
+
+### Register an API Key
+
+API key registration allows you to secure their APIs that are behind an auth and use in message extensions. You can register an API key and specify the domain, tenant, and app that can access the APIs, and provide the secrets that are needed to authenticate the API calls.  You can then paste the API key ID in the simplified messaging extension UI to enable the authentication.
+
+To register an API Key, follow these steps:
+
+1. Go to **Tools** > **API Key Registration**.
+
+1. Select **New API Key**.
+
+1. In the **Register an API key** page, update the following:
+
+   1. **Description**: Description of the API Key.
+   1. **Add Domain**: The domain where you host the API.
+
+1. Under **Set a target tenant**, select the following:
+   1. If you want to use the API key to call the APIs for a specific tenant, select **Home tenant**.
+   1. If you want to use the API key to call the APIs for any tenant, select **Any tenant**.
+
+1. Under **Set a Teams app**, select the following:
+   1. If you want the app that matches the given teams App ID to use the API key, select **Existing Teams app**.
+   1. If you want the app that matches any teams App ID to use the API key, select **Any Teams app**.
+
+1. Select **+ Add Secret** and enter the OpenAI API secret key.
+
+1. Select **Save**. An **API key registration ID** is generated.
+
+# [Microsoft Entra ID](#tab/microsoft-entra-id)
+
+`microsoftEntra` authentication method uses an app user's Teams identity to provide them access to your app. A user who has logged into Teams doesn't need to log in again to your app within the Teams environment. With only a consent required from the app user, the Teams app retrieves access details for them from Microsoft Entra ID. After the app user has given consent, they can access the app even from other devices without having to be validated again.
+
+### Prerequisites
+
+Before you start, ensure you have the following:
+
+* An Azure account with an active subscription.
```

Comment:
Can we align the name in this doc and replace all Azure AD/AAD with Microsoft Entra?